win32ss/user/ntuser/kbdlayout.c

```diff
@@ -1082,6 +1105,8 @@ NtUserLoadKeyboardLayoutEx(
             pKl->pklPrev = pKl;
             gspklBaseLayout = pKl;
         }
+
+        pKl->piiex = co_UserImmLoadLayout(UlongToHandle(hkl));
```

Comment:
Note: this `co_UserImmLoadLayout` call can return NULL (see your implementation); however, you continue execution afterwards in `NtUserLoadKeyboardLayoutEx` without doing anything more. I therefore wonder whether, in this case, your new keyboard-layout may be in a somewhat inconsistent state (in case `pKl->piiex` being NULL is not valid... I don't know).

Because later, when that layout gets unloaded (see code above), you unconditionally free the `pKl->piiex` with `ExFreePoolWithTag`, which may be NULL, as explained above. In this case you may end up with a BSoD. If `pKl->piiex` being NULL may be expected for a keyboard layout, then in that cleanup path, you need to check for NULL before calling `ExFreePoolWithTag`.

Reply:
NULL check added.